protocol/storage/miner.go

```diff
@@ -580,17 +579,12 @@ func (sm *Miner) onCommitFail(ctx context.Context, dealCid cid.Cid, message stri
 		resp.Message = message
 		resp.State = storagedeal.Failed
 	})
-	log.Errorf("commit failure but could not update to deal 'Failed' state: %s", err)
+	sm.log.Errorf("commit failure but could not update to deal 'Failed' state: %s", err)
 }
 
 // currentProvingPeriodPoStChallengeSeed produces a PoSt challenge seed for
 // the miner actor's current proving period.
-func (sm *Miner) currentProvingPeriodPoStChallengeSeed(ctx context.Context) (types.PoStChallengeSeed, error) {
-	currentProvingPeriodStart, err := sm.getProvingPeriodStart()
-	if err != nil {
-		return types.PoStChallengeSeed{}, errors.Wrap(err, "error obtaining current proving period")
-	}
-
+func (sm *Miner) currentProvingPeriodPoStChallengeSeed(ctx context.Context, currentProvingPeriodStart *types.BlockHeight) (types.PoStChallengeSeed, error) {
```

Comment:
STRONGLY-WORDED SUGGESTION: This method's name should be changed to `poStChallengeSeedAtBlockHeight`, or you should drop the `currentProvingPeriodStart` parameter and instead, in the body of the method, make a call to `getProvingPeriod`.